### PR TITLE
Fix #17, make the browse message target only a specific window

### LINKED ISSUE #17: Make it work in multiple windows
While the sidebar is opened in the "current" window, the message that is sent goes to all sidebars. Additionally, when opening a new sidebar in a new window, the retry code stops retrying as soon as it contacts any sidebar, which could be a sidebar in an unintended window.

It might help if we set the URL to `sidebar.html#url={url_to_open}` instead of using the messaging.

### PR DESCRIPTION
**Pull request by ianb**
Actually this has turned into a miscellaneous set of commits.

### CODE REVIEW

File: addon/background.js [hunk 1]
```diff
@@ -38,8 +54,33 @@ let requestFilter = {
   urls: ["http://*/*", "https://*/*"],
 };
 
+let desktopHostnames = {};
```
[the remark this answers is not in the document]
Comment (ianb) [reply addon/background.js]: Correct, it has to be JSON-compatible

File: addon/sidebar.js [hunk 2]
```diff
@@ -1,22 +1,52 @@
...
   element("#browser-iframe").src = url;
   let urlObj = new URL(url);
   element("#browser-domain").textContent = urlObj.hostname;
+  element("#desktop").checked = !!desktop;
 }
 
 function displayHome() {
   element("#browser-container").style.display = "none";
   element("#onboarding").style.display = "";
+  browser.windows.getCurrent({populate: true}).then((windowInfo) => {
```
[the remark this answers is not in the document]
Comment (ianb) [reply addon/sidebar.js]: oh yeah, I already opened #19 for this

File: addon/background.js [hunk 3]
```diff
@@ -24,11 +24,27 @@ browser.contextMenus.create({
 browser.contextMenus.onClicked.addListener((info, tab) => {
   let url = info.linkUrl || tab.url;
   let title = info.linkText || tab.title || "Page";
-  sidebarOpenAndSend({type: "browse", url}).catch((error) => {
+  browser.sidebarAction.open().then(() => {
```
[the remark this answers is not in the document]
Comment (ianb) [reply addon/background.js]: I like the consistency of the layout by using a block each time. Though I think we could use async/await...?

File: addon/background.js [hunk 4]
```diff
@@ -24,11 +24,27 @@ browser.contextMenus.create({
...
+    return browser.windows.getCurrent();
+  }).then((windowInfo) => {
+    let desktop = !!desktopHostnames[(new URL(url)).hostname];
```
[the remark this answers is not in the document]
Comment (ianb) [reply addon/background.js]: Yeah, the filtering happens globally too, so it's hard to target a tab in any consistent way. Currently I think a tab that doesn't get reloaded, but has a hostname whose desktop property is changed, will not itself get loaded. Kind of a corner case though.

File: addon/background.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (fzzzy) [reply addon/background.js]: Sure, I totally understand that. That's fine.

**Pull request merged**